File: app/src/main/java/com/conoland/pokedex/Detalles_PojkemonAdapter.java
package com.conoland.pokedex;
import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.conoland.pokedex.models.Pokemon;

import java.util.ArrayList;

public class Detalles_PojkemonAdapter extends RecyclerView.Adapter<Detalles_PojkemonAdapter.ViewHolder> {

    private ArrayList<Pokemon>listaDetalles = new ArrayList<>();
    private Context context;
    private ListaPokemonAdapter parent;

    public Detalles_PojkemonAdapter(Context context, ListaPokemonAdapter parent) {
        this.context = context;
        this.parent = parent;
    }

    public  void adicionarDetallesPokemon(ArrayList<Pokemon> list) {
        listaDetalles.addAll(list);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public Detalles_PojkemonAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_etails_pokemon,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull Detalles_PojkemonAdapter.ViewHolder holder, int position) {
    Pokemon p = listaDetalles.get(position);
    holder.name.setText(p.getName());
    holder.weight.setText("PESO: "+p.getWeight()+"Kg");
    Glide.with(holder.name.getContext()).load("https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/"+p.getNumber()+".png")
            .centerCrop().diskCacheStrategy(DiskCacheStrategy.ALL).into(holder.img);
    }

    @Override
    public int getItemCount() {
        return listaDetalles.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        ImageView img;
        TextView name;
        TextView weight;
        TextView volver;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            img= itemView.findViewById(R.id.fotoD);
            name= itemView.findViewById(R.id.nombreD);
            weight= itemView.findViewById(R.id.pesoD);
            volver=itemView.findViewById(R.id.volver);

           volver.setOnClickListener(view->{
               Activity activity=((Activity) context);
               RecyclerView recyclerView = activity.findViewById(R.id.recyclerView);
               final GridLayoutManager layoutManager = new GridLayoutManager(context, 2);
               recyclerView.setLayoutManager(layoutManager);
               recyclerView.setAdapter(parent);


           });
        }
    }
}

File: app/src/main/java/com/conoland/pokedex/Detalles_PojkemonAdapter.kt
package com.conoland.pokedex

import android.app.Activity
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.conoland.pokedex.models.Pokemon

class Detalles_PojkemonAdapter(
    private val context: Context,
    private val parent: ListaPokemonAdapter
) :
    RecyclerView.Adapter<Detalles_PojkemonAdapter.ViewHolder>() {
    private val listaDetalles = ArrayList<Pokemon>()

    fun adicionarDetallesPokemon(list: ArrayList<Pokemon>) {
        listaDetalles.addAll(list)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.item_etails_pokemon, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val p = listaDetalles[position]
        holder.name.text = p.name
        holder.weight.text = "PESO: " + p.weight + "Kg"
        Glide.with(holder.name.context)
            .load("https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/" + p.number + ".png")
            .centerCrop().diskCacheStrategy(DiskCacheStrategy.ALL).into(holder.img)
    }

    override fun getItemCount(): Int {
        return listaDetalles.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var img: ImageView =
            itemView.findViewById(R.id.fotoD)
        var name: TextView = itemView.findViewById(R.id.nombreD)
        var weight: TextView = itemView.findViewById(R.id.pesoD)
        var volver: TextView = itemView.findViewById(R.id.volver)

        init {
            volver.setOnClickListener { view: View? ->
                val activity = (context as Activity)
                val recyclerView =
                    activity.findViewById<RecyclerView>(R.id.recyclerView)
                val layoutManager = GridLayoutManager(context, 2)
                recyclerView.layoutManager = layoutManager
                recyclerView.adapter = parent
            }
        }
    }
}
